docs/libssh2_session_callback_set.3

```diff
@@ -71,8 +71,65 @@ to the abstract pointer set in the \fIlibssh2_session_init_ex(3)\fP call.
 The callback returns the number of bytes read, or -1 for error. The special
 return code \fB-EAGAIN\fP can be returned to signal that the read was aborted
 to prevent getting blocked and it needs to be called again.
+.IP LIBSSH2_CALLBACK_AUTHAGENT
+Called during authentication process to allow the client to connect to the
+ssh-agent and perform any setup, such as configuring the agent or adding keys.
+
+The prototype of the callback:
+
+.nf
+void authagent(LIBSSH2_SESSION* session, LIBSSH2_CHANNEL *channel,
+               void **abstract);
+.fi
+.IP LIBSSH2_CALLBACK_AUTHAGENT_IDENTITIES
+Not called by libssh2. The client is responsible for calling this method when
+a SSH2_AGENTC_REQUEST_IDENTITIES message has been received.
+
+The prototype of the callback:
+
+.nf
+void identities(LIBSSH2_SESSION* session, void *buffer,
+                const char *agent_path,
+                void **abstract)
+.fi
+
+\fBbuffer\fP must be filled in by the callback. Different clients may implement
+this differently. For example, one client may pass in an unsigned char ** for
+this parameter, while another may pass in a pointer to a struct.
+
+Regardless of the type of buffer used, the client will need to send back a list
+of identities in the following format.
+
+uint32 buffer length
+uint32 number of entries
+entries
+
+Where each entry in the entries list is of the format:
+
+string data
+cstring comment
+
+\fBagent_path\fP The path to a running ssh-agent on the client machine, from
+which identities can be listed.
+.IP LIBSSH2_CALLBACK_AUTHAGENT_SIGN
+Not called by libssh2. The client is responsible for calling this method when
+a SSH2_AGENTC_SIGN_REQUEST message has been received.
+
+The prototype of the callback:
+
+.nf
+void sign(LIBSSH2_SESSION *session,
+          unsigned char **sig, size_t *sig_len,
+          const unsigned char *data, size_t data_len,
+          void **abstract);
```

Comment:
The callback arguments here (`LIBSSH2_CALLBACK_AUTHAGENT_IDENTITIES` and `LIBSSH2_CALLBACK_AUTHAGENT_SIGN`) seem to differ from the ones in the code, or am I misreading?

For example from the SIGN callback, blob seems to be missing and `sign` comes before `data` here, but after in the code.

Reply:
You're right about `LIBSSH2_CALLBACK_AUTHAGENT_SIGN`. I think it was changed after I last edited the man page.

It should be `int sign(LIBSSH2_SESSION* session, unsigned int flags, unsigned char *blob, unsigned int blen, unsigned char *data, const unsigned int dlen, unsigned char **sig, unsigned int *sig_len, const char *agent_path, void **abstract);`

I'll update the man page in a bit here.

`LIBSSH2_CALLBACK_AUTHAGENT_IDENTITIES` looks correct to me though, unless I'm missing something.

Reply:
`sign()` will be fine then, thanks for looking into it.

As for `identities`, I'm looking at this code in `include/libssh2.h`:
```
#define LIBSSH2_ADD_IDENTITIES_FUNC(name) \
   void name(LIBSSH2_SESSION *session, LIBSSH2_CHANNEL *channel, \
             void **abstract)
```
This doesn't match `LIBSSH2_CALLBACK_AUTHAGENT_IDENTITIES` in 
the manual or `LIBSSH2_ADD_IDENTITIES()` from `libssh2_priv.h`. They 
look like the same callback, though I may be wrong.

Reply:
Oops, I was looking at the wrong branch. The identities in the man file was correct, but it was incorrect in the header. I was looking at a branch where it was correct in the header.

Both problems should now be fixed. The man file was updated with the correct `sign()`, and the header was updated to match `identities()` in the man file.

Reply:
Thanks, no worries. We're good to merge once the CI tests finished.